Modix.Services/Core/MessageLogBehavior.cs

```diff
@@ -54,6 +62,44 @@ private async Task<bool> ShouldSkip(IGuild guild, IMessageChannel channel)
             return result;
         }
 
+        private async Task HandleMessageReceived(SocketMessage message)
+        {
+            Log.LogDebug("Handling message received event for message #{MessageId}.", message.Id);
+
+            if (!message.Content.StartsWith('!') &&
+                message.Channel is IGuildChannel channel &&
+                message.Author is IGuildUser author &&
+                !author.IsBot && !author.IsWebhook)
```

Comment:
checking against `IGuildUser` and `IGuildChannel` are 100% not sufficient to check if there is actually a guild value. The only surefire way is to do the full cast and check whether `Guild` is `null`.

Reply:
I don't actually access a `Guild` property. I use `IGuildChannel.GuildId` which is of type `ulong` (not `ulong?`).

Reply:
Good to know. I'll fix it.